File: angular/src/app/model/chat.dto.ts
export interface ChatDTO {
id: string;
name: string;
userId: string;
userName: string;
}

export interface ChatView {
  chatId: string;
  chatName: string;
  userId: string;
  userName: string;
}

export interface ChatMessageView {
  id: string;
  chatId: string;
  userId: string;
  userName: string;
  messageText: string;
  isNew: boolean;
  timestamp: string;
}

export class ChatMessageForm {
  constructor(
    public message: string
  ){}
}


export interface MessageHubModel {
  userId: string;
  chatName: string;
  message: string;
}

File: webapi/OtusWebApi/DataClasses/Dtos/ChatDto.cs
namespace OtusSocialNetwork.DataClasses.Dtos
{
    public record ChatDTO(string Id, string Name, string UserId, string UserName)
    {

    }

    public record ChatForm(string UserId);
    public record ChatMessageForm(string Message);

    public record ChatItem(string Id, string Name, string[] UserIds);
    public record ChatSocket(string UserId, string ConnectionId);
    public record ChatMessageItem(string Id, string ChatId, string UserId, string Message, string Timestamp );
}
